src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private int capacity;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+    }
 
     @Override
     public void put(K key, V value) {
-
+        table = resize();
+        int hash = (key == null) ? 0 : key.hashCode();
+        int bucket = Math.abs(hash) % capacity;
+        Node<K, V> node = table[bucket];
+        Node<K, V> newNode;
+        if (node == null) {
+            table[bucket] = new Node<>(hash,key,value, null);
+            size++;
+            return;
+        }
+        while (node != null) {
+            if (hash == node.hash && node.key == null ? key == null : node.key.equals(key)) {
```

Comment:
```suggestion
            if (hash == node.hash && node.key == null ? key == null : node.key.equals(key)) {
```
I can't undersend this. Can you simplify second part ?